src/libstd/sys/windows/net.rs

```diff
@@ -131,9 +132,9 @@ impl Socket {
     pub fn read(&self, buf: &mut [u8]) -> io::Result<usize> {
         // On unix when a socket is shut down all further reads return 0, so we
         // do the same on windows to map a shut down socket to returning EOF.
+        let len = cmp::min(buf.len(), i32::max_value()) as i32;
```

Comment:
This is missing an `as usize`.
May as well make this line match the others:
`let len = cmp::min(buf.len(), <wrlen_t>::max_value() as usize) as wrlen_t;`